src/types/Token.ts

```diff
@@ -1,10 +1,19 @@
-interface Token {
+type CustomTokenPart<T> = {
+  [P in keyof T]: T[P]
+};
+
+type StaticTokenPart = {
   access_token: string;
   expires_in?: number;
   scope?: string[];
   token_type?: string;
   local_expiry?: number;
-  [key: string]: any;
 }
 
-export { Token };
+type Token<T = any> = CustomTokenPart<T> & StaticTokenPart;
```

Comment:
@bzums I just checked what happens if we would omit the default type, but then we would have to adjust several pieces, as they are not prepared to handle this generic stuff.

@bzums @Retro64 @ISO50 Thus, I removed the unnecessary type declaration. I thinks it's now ready to be reviewed 👍